File: Beakjoon/_etc/Java/9461 파도반 수열.java

import java.util.Scanner;

public class Main {
    public static void main(String args[]) {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt();
      long[] dp = new long[101];
      
      dp[0] = 0;
      dp[1] = 1;
      dp[2] = 1;
      dp[3] = 1;
      dp[4] = 2;
      dp[5] = 2;
      
      for(int i = 6; i < 101; i++) {
          dp[i] = dp[i-1] + dp[i-5];
      }
      
      for(int i = 0; i < n; i++) {
          
          int num = s.nextInt();
          System.out.println(dp[num]);
          
      }
      
    }
}
// http://boj.kr/65c0f959215f4e6a9e216b38b6612236

File: _etc/Kotlin/9461 파도반 수열.kt
import java.util.Scanner
import kotlin.math.*

fun main(args: Array<String>) {
    val sc: Scanner = Scanner(System.`in`)
    val num = sc.nextInt()
    
    var dp = LongArray(101, {0})
    
    dp[0] = 0
    dp[1] = 1
    dp[2] = 1
    dp[3] = 1
    dp[4] = 2
    dp[5] = 2
    
    for(i in 6..100) {
        dp[i] = dp[i-1] + dp[i-5]
    }
    
    for(i in 0 until num) {
        var n = sc.nextInt()
        println("${dp[n]}")
    }
}

// http://boj.kr/1ac6b9c233804746b35432f14a334237
